slang-kotlin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -115,10 +116,13 @@ private Tree createElement(@Nullable PsiElement element) {
     } else if (element instanceof KtConstantExpression) {
       return new LiteralTreeImpl(metaData, element.getText());
     } else if (element instanceof KtBlockExpression) {
-      List<Tree> statementOrExpressions = ((KtBlockExpression) element).getStatements().stream().map(this::createElement).collect(Collectors.toList());
+      List<Tree> statementOrExpressions = ((KtBlockExpression) element).getStatements().stream()
+        .map(this::createElement)
+        .filter(Objects::nonNull)
```

Comment:
It's not obvious why we are filtering nonNull objects here and same for the rest of this method. I suggest we add a comment at the beginning of this method explaining this.